sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,472 @@
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_load(value=-3, start=0, order=-2)
+        >>> b.apply_load(value=4, start=2, order=-1)
+        >>> b.apply_load(value=-2, start=2, order=2, end = 3)
+        >>> b.load
+        -3*SingularityFunction(x, 0, -2) + 4*SingularityFunction(x, 2, -1) - 2*SingularityFunction(x, 2, 2)
+            + 2*SingularityFunction(x, 3, 0) + 2*SingularityFunction(x, 3, 2)
+        """
+        x = self.variable
+        value = sympify(value)
+        start = sympify(start)
+        order = sympify(order)
+
+        self._load += value*SingularityFunction(x, start, order)
+
+        if end:
+            if order == 0:
+                self._load += -value*SingularityFunction(x, end, order)
+            elif order.is_positive:
+                self._load += -value*SingularityFunction(x, end, order) - value*SingularityFunction(x, end, 0)
+            else:
+                raise ValueError("""Order of the load should be positive.""")
+
+    @property
+    def load(self):
+        """
+        Returns a Singularity Function expression which represents
+        the load distribution curve of the Beam object.
+
+        Examples
+        ========
+        There is a beam of length 4 meters. A moment of magnitude 3 Nm is
+        applied in the clockwise direction at the starting point of the beam.
+        A pointload of magnitude 4 N is applied from the top of the beam at
+        2 meters from the starting point and a parabolic ramp load of magnitude
+        2 Nm/m is applied below the beam starting from 3 meters away from the
+        starting point of the beam.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_load(value=-3, start=0, order=-2)
+        >>> b.apply_load(value=4, start=2, order=-1)
+        >>> b.apply_load(value=-2, start=3, order=2)
+        >>> b.load
+        -3*SingularityFunction(x, 0, -2) + 4*SingularityFunction(x, 2, -1) - 2*SingularityFunction(x, 3, 2)
+        """
+        return self._load
+
+    def shear_force(self):
+        """
+        Returns a Singularity Function expression which represents
+        the shear force curve of the Beam object.
+
+        Examples
+        ========
+        There is a beam of length 4 meters. A moment of magnitude 3 Nm is
+        applied in the clockwise direction at the starting point of the beam.
+        A pointload of magnitude 4 N is applied from the top of the beam at
+        2 meters from the starting point and a parabolic ramp load of magnitude
+        2 Nm/m is applied below the beam starting from 3 meters away from the
+        starting point of the beam. The bending moment at 0 should be 4 and
+        at 4 it should be 0. The slope of the beam should be 1 at 0. The
+        deflection should be 2 at 0.
```

Comment:
It isn't clear how you specify reaction loads at any supports in these examples. It is odd to try to think of a beam that is floating in space, as it is not a static thing anymore.